### PR TITLE
[release-4.7] Bug 1995340: Fix metrics endpoint + Update golden image + IBM cluster support

### PR DESCRIPTION
**Pull request by jrvaldes**
This PR combines a few backports that have inter-dependencies.

While backporting the updates for the golden image template in vSphere, [Bug 1995340](https://bugzilla.redhat.com/show_bug.cgi?id=1995340) was identified and resolved. 
[Bug 2008601 ](https://bugzilla.redhat.com/show_bug.cgi?id=2008601) is a follow-up fix targeting 4.10, that is manually cherry-picked.

The support for IBMCloud is a dependency, and also got manually cherry-picked.

### CODE REVIEW

**Review by aravindhp — APPROVED** [review]
/approve

**Comment by aravindhp** [comment]
/retitle [release-4.7][e2e] Update golden image template for vSphere

**Comment by jrvaldes** [comment]
/test vsphere-e2e-operator

retest due to network issues.

**Comment by aravindhp** [comment]
/approve cancel

@jrvaldes please do not mass backport unrelated commits

**Comment by jrvaldes** [comment]
> /approve cancel
> 
> @jrvaldes please do not mass backport unrelated commits

The goal is to bring https://github.com/openshift/windows-machine-config-operator/commit/6fcb30f527b499fd3d6ddc0550d38aaa68ae760e , since auto cherry-picking [failed](https://github.com/openshift/windows-machine-config-operator/pull/544#issuecomment-897889392). Good catch, dropped the [unrelated commit](https://github.com/openshift/windows-machine-config-operator/commit/fa141f2b401a4a81963194239b0d095ed258693f), I was testing locally with it and got pushed by mistake.

**Comment by aravindhp** [comment]
> > /approve cancel
> > @jrvaldes please do not mass backport unrelated commits
> 
> The goal is to bring [6fcb30f](https://github.com/openshift/windows-machine-config-operator/commit/6fcb30f527b499fd3d6ddc0550d38aaa68ae760e) , since auto cherry-picking [failed](https://github.com/openshift/windows-machine-config-operator/pull/544#issuecomment-897889392).

Do you know why the auto cherry-pick failed? Is it because you are doing out of order cherry-picking? Please ensure all the required changes are being back ported in the correct order.

**Comment by jrvaldes** [comment]
/test vsphere-e2e-operator

https://github.com/openshift/windows-machine-config-operator/pull/614 merged.

**Comment by jrvaldes** [comment]
/test vsphere-e2e-operator

timed out waiting for creation

**Comment by jrvaldes** [comment]
/test vsphere-e2e-operator

Failed creating infrastructure resources

**Comment by jrvaldes** [comment]
/test vsphere-e2e-operator

Landed in IBMCloud

**Review by aravindhp — APPROVED** [review]
/approve

**Review by aravindhp — APPROVED** [review]
/lgtm

**Comment by aravindhp** [comment]
Seeing `E0924 18:45:22.931085       1 actuator.go:57] e2e-wm-p2ccw error: e2e-wm-p2ccw: reconciler failed to Create machine: Permission to perform this operation was denied.` in `vsphere-e2e-operator ` test. ref: [machine-controller log](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/538/pull-ci-openshift-windows-machine-config-operator-release-4.7-vsphere-e2e-operator/1441457956649439232/artifacts/vsphere-e2e-operator/gather-extra/artifacts/pods/openshift-machine-api_machine-api-controllers-5778bd4d88-q9kbq_machine-controller.log)

/retest-required

**Comment by aravindhp** [comment]
/hold

Continuing to see permission issues on IBM cloud cluster
```
E0925 16:08:52.968417       1 actuator.go:57] e2e-wm-bnfbw error: e2e-wm-bnfbw: reconciler failed to Create machine: Permission to perform this operation was denied.
```

**Comment by aravindhp** [comment]
/hold cancel
/retest

The permission issue on the IBM cluster has been fixed

**Comment by aravindhp** [comment]
/hold

until we decide if we need to force merge or add the bug fix to this PR

**Review by aravindhp — CHANGES_REQUESTED** [review]
As mentioned during the discussion, this PR has to be reframed / retitled so that it is associated with the release-4.7 bug.

**Review by aravindhp — APPROVED** [review]
/approve
/retest

**Pull request merged**
